src/renderer/coremods/welcome/index.ts

```diff
@@ -0,0 +1,9 @@
+import { generalSettings } from "../settings/pages/General";
+import { notices } from "@replugged";
+import { Messages } from "@common/i18n";
+
+export function start(): void {
+  if (!generalSettings.get("showWelcomeNoticeOnOpen")) return;
+  notices.sendAnnouncement({ message: Messages.REPLUGGED_WELCOME_NOTICE });
```

Comment:
We already have two i18n strings: `REPLUGGED_NOTICES_WELCOME_NEW_USER` and `REPLUGGED_NOTICES_JOIN_SERVER_BUTTON`.
Add an onClose prop that sets `showWelcomeNoticeOnOpen` to `false`, and a `button` prop that lets you join our server (we have a util function `goToOrJoinServer` for this).